### ISSUE TITLE
Experimenting with TypeDoc

### ISSUE THREAD

**Issue by Bartvds** [DefinitelyTyped/docs#1]
The idea is to let Travis generate API docs from the definitions.

I'm running some experiments with https://github.com/sebastian-lenz/typedoc

I put it in it's own repo so we can run it on Travis: TypeDoc is based on the TSC compiler, and it is very thorough but pretty slow and we need the multi-core (my crappy PC is throwing blue-smoke so let somebody else pay for cycles) :wink:

@dt-bot can then push the result to the gh-pages branch so it ends up as http://definitelytyped.org/docs (if anybody got a better name then let me know).

It's not quite there yet but I'm pretty hopeful this will work out.

**Comment by Bartvds** [comment]
I accidentally the whole repo.. 

But it is epic:

http://definitelytyped.org/docs/ (not linked from site, just for fun & testing)

It took Travis 13 minutes on 24 cores/threads to bake 531 pages, nicely inefficient with duplicate assets and everything.

https://github.com/DefinitelyTyped/docs/tree/gh-pages

Fun times. Now I got to learn SASS and Handlerbars.. :zap::icecream::zap:

**Comment by borisyankov** [comment]
It looks double awesome.

And SASS is easy. It's just CSS + awesome.

You just need to remember it is not actually a programming language. ;)

**Comment by Bartvds** [comment]
Yes, well, it is an existing SASS & template system written by a programmer :smile: https://github.com/sebastian-lenz/typedoc/tree/master/themes/default/assets/css

Not sure if that is good or bad. At least it looks tidy.

**Comment by sebastian-lenz** [comment]
Actually I'm a graphics designer ;) But I have to admit the current template is not in best shape. Let me know if I can help or if you run into problems.

I've also seen you are looking for help on the DefinitelyTyped page somewhere. Maybe I could contribute somehow, what do you plan/need?

**Comment by Bartvds** [comment]
@sebastian-lenz Oh, your code for TypeDoc looked very programmer wise with a real architecture and everything.

Current 'design' of for dt.org and tsdpm.com is developer art by myself hacked from some templates. We're supposedly have a designer; there are some mocks for DT site but I have no idea about status (it is all spare time work). I guess we're sort of OK until that becomes a bit more real.

And content, always more content. Directory and Guides needs some but nobody has time for that.
